Sumar del 1 al 100:

$$
1 + 2 + 3 + ... + 97 + 98 + 99 + 100
$$

In [3]:
suma = 0

for i in range(1, 101):
    suma = suma + i

print(suma)

5050


La suma de los pares del 1 al 100:

$$
2 + 4 + 6 + 8 + ... + 96 + 98 + 100
$$

In [5]:
# i % 2 significa i MOD 2, que es el residuo entero de quitar 2 a i,
# 1897 % 2, 948 + residuo de 1, 1987 % 2 -> 1

suma = 0

for i in range(1, 101):
    if i % 2 == 0: # Par
        suma = suma + i

suma

2550

$$
1 + 2 + 3 + ... + (n - 1) + n = \frac{n \cdot (n + 1)}{2}
$$

$$
1 + 2 + 3 + ... + 99 + 100 = \frac{100 \cdot 101}{2} = \frac{10100}{2} = 5050
$$

$$
2 + 4 + 6 + ... + 98 + 100 = 2 \cdot (1 + 2 + 3 + ... + 49 + 50)
$$

$$
2 + 4 + 6 + ... + 98 + 100 = 2 \cdot \frac{50 \cdot 51}{2} = 50 \cdot 51
$$

$$
2 + 4 + 6 + ... + 98 + 100 = 50 \cdot 51 = 2550
$$